tools/protoxform_test_helper.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python3
+
+from run_command import runCommand
+
+import logging
+import os
+import re
+import sys
+
+
+def PathAndFilename(label):
+  """Retrieve actual path and filename from bazel label
+
+  Args:
+    label: bazel label to specify target proto.
+
+  Returns:
+    actual path and filename
+  """
+  if label.startswith('/'):
+    label = label.replace('//', '/', 1)
+  elif label.startswith('@'):
+    label = re.sub(r'@.*/', '/', label)
+  else:
+    return label
+  label = label.replace(":", "/")
+  splitted_label = label.split('/')
+  return ['/'.join(splitted_label[:len(splitted_label) - 1]), splitted_label[-1]]
+
+
+def GoldenProtoFile(path, filename, version):
+  """Retrieve golden proto file path. In general, those are placed in tools/testdata/protoxform.
+
+  Args:
+    path: target proto path
+    filename: target proto filename
+    version: api version to specify target golden proto filename
+
+  Returns:
+    actual golden proto absolute path 
+  """
+  base = "./"
+  base += path + "/" + filename + "." + version + ".gold"
+  return os.path.abspath(base)
+
+
+def ResultProtoFile(path, filename, version):
+  """Retrieve result proto file path. In general, those are placed in bazel artifacts.
+
+  Args:
+    path: target proto path
+    filename: target proto filename
+    version: api version to specify target result proto filename
+
+  Returns:
+    actual result proto absolute path
+  """
+  base = "./bazel-bin/"
+  delimited_path = path.split('/')
+  base += os.path.join(*delimited_path, "protos", *delimited_path)
```

Comment:
I think this might be where the pyformat failure is happening; how come you want to squash the path twice here and join it twice? Why not just use `path` instead of `delimited_path`?

Reply:
@htuch automatically generated proto file is located on `./bazel-bin/tools/testdata/protoxform/envoy/v2/protos/tools/testdata/protoxform/envoy/v2/`.
So, this instruction could not avoid. But, It seems for me to use `path` instead of `delimited_path`.